src/main/native/cpp/analysis/JSONConverter.cpp

```diff
@@ -84,10 +89,68 @@ std::string sysid::ConvertJSON(wpi::StringRef path, wpi::Logger& logger) {
 
   // Write the new file with "_new" appended to it.
   std::string loc = path.rsplit(".json").first.str() + "_new.json";
+  std::error_code ec;
   wpi::raw_fd_ostream output{loc, ec};
   output << json;
   output.flush();
 
   WPI_INFO(logger, "Wrote new JSON to: " << loc);
   return loc;
 }
+
+std::string sysid::ToCSV(wpi::StringRef path, wpi::Logger& logger) {
+  wpi::json json = GetJSON(path, logger);
+
+  auto type = sysid::analysis::FromName(json.at("test").get<std::string>());
+  auto factor = json.at("unitsPerRotation").get<double>();
+  auto unit = json.at("units").get<std::string>();
+
+  std::error_code ec;
+  // Naming: {sysid-json-name}(Test, Units).csv
+  std::string loc = path.rsplit(".json").first.str() + "(" + type.name + ", " +
+                    unit + ").csv";
+  wpi::raw_fd_ostream outputFile{loc, ec};
+
+  if (ec) {
+    throw std::runtime_error("Unable to write to: " + loc);
+  }
+
+  outputFile << "Timestamp, Test, ";
+  if (type == analysis::kDrivetrain || type == analysis::kDrivetrainAngular) {
+    outputFile << "Left Volts, Right Volts, Left Position, Right Position, "
```

Comment:
This adds invisible spaces to the CSV fields. Generally, the CSV format has these stripped.

Reply:
is there a way to get the unit abbreviation or should I just do a bunch of ifs?

Reply:
If we use the units library, we can grab the string suffix programmatically. That's not how the code's currently set up though.

Reply:
or could we just append the entire unit name?

Reply:
We could, but engineering convention for plot axis labels uses the name of the quantity followed by the unit abbreviation in parentheses.

Reply:
So if statements would be the way to go?

Reply:
Yea, that would be easiest.